internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -0,0 +1,755 @@
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: r.Namespace,
+			Labels: map[string]string{
+				"app": name,
+			},
+		},
+	}
+	svcSpec := corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			{
+				Name:       name,
+				Port:       servicePort,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(int(servicePort)),
+			},
+		},
+		Selector: map[string]string{
+			"app": name,
+		},
+		Type: corev1.ServiceTypeLoadBalancer,
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, svc, r.Scheme); err != nil {
+			return err
+		}
+		svc.Spec = svcSpec
+		return nil
+	}
+
+	return svc, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newPostgresService(instance *adiiov1alpha1.AgentServiceConfig) (*corev1.Service, controllerutil.MutateFn) {
+	svc := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      postgresDeploymentName,
+			Namespace: r.Namespace,
+			Labels: map[string]string{
+				"app": postgresDeploymentName,
+			},
+		},
+	}
+	svcSpec := corev1.ServiceSpec{
+		Ports: []corev1.ServicePort{
+			{
+				Name:       postgresDeploymentName,
+				Port:       databasePort,
+				Protocol:   corev1.ProtocolTCP,
+				TargetPort: intstr.FromInt(int(databasePort)),
+			},
+		},
+		Selector: map[string]string{
+			"app": postgresDeploymentName,
+		},
+		Type: corev1.ServiceTypeLoadBalancer,
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, svc, r.Scheme); err != nil {
+			return err
+		}
+		svc.Spec = svcSpec
+		return nil
+	}
+
+	return svc, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newAgentRoute(instance *adiiov1alpha1.AgentServiceConfig) (*routev1.Route, controllerutil.MutateFn) {
+	weight := int32(100)
+	route := &routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: r.Namespace,
+		},
+	}
+	routeSpec := routev1.RouteSpec{
+		To: routev1.RouteTargetReference{
+			Kind:   "Service",
+			Name:   name,
+			Weight: &weight,
+		},
+		Port: &routev1.RoutePort{
+			TargetPort: intstr.FromString(name),
+		},
+		WildcardPolicy: routev1.WildcardPolicyNone,
+	}
+
+	mutateFn := func() error {
+		if err := controllerutil.SetControllerReference(instance, route, r.Scheme); err != nil {
+			return err
+		}
+		route.Spec = routeSpec
+		return nil
+	}
+
+	return route, mutateFn
+}
+
+func (r *AgentServiceConfigReconciler) newPostgresSecret(instance *adiiov1alpha1.AgentServiceConfig) (*corev1.Secret, controllerutil.MutateFn) {
+	secret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      databaseSecretName,
+			Namespace: r.Namespace,
+		},
+	}
+
+	// TODO(djzager): Use legit password
+	secretData := map[string]string{
+		"db.host":     databaseName,
+		"db.user":     "admin",
+		"db.password": "abcdefg",
```

Comment:
Need to add special characters to make it secure. j/k ;)

Reply:
Thanks for the reminder. This should be addressed before merge.